products/green_fast_track/models/intermediate/flags/int__zoning_districts.sql

```diff
@@ -0,0 +1,37 @@
+WITH pluto AS (
+    SELECT * FROM {{ ref('stg__pluto') }}
+),
+
+districts_exploded AS (
+    SELECT
+        bbl,
+        UNNEST(ARRAY[zonedist1, zonedist2, zonedist3, zonedist4]) AS zd,
+        zonedist1 IS null AS zd_all_null
+    FROM pluto
+),
+
+districts_distinct AS (
+    SELECT
+        bbl,
+        zd
+    FROM districts_exploded
+    WHERE (zd IS NOT null AND NOT zd_all_null) OR zd_all_null
+    GROUP BY 1, 2
```

Comment:
Nit - group/order by column names rather than numbers?
Also since we're not actually aggregating here, this one could just be a `SELECT DISTINCT` instead of a group by

Reply:
And actually, pluto build ensures that these fields are distinct from each other, so the distinct isn't needed in this CTE I don't think

Reply:
Well other than the all nulls I suppose. Hmm

Reply:
For districts exploded, could do

```sql
SELECT
    bbl,
    unnest(CASE 
        WHEN zonedist1 IS NULL THEN ARRAY['NONE']
        ELSE ARRAY[zonedist1, zonedist2, zonedist3, zonedist4]
    end) AS zd
FROM pluto
```

And then after that, only have to filter `WHERE zd IS NOT NULL`

Reply:
since Jack is using this branch's build export and this being a nit, let's save this for later

Reply:
If by later, you mean still before merging!

Reply:
Edited above query slightly to make even simpler

Really, it might make most sense to put that `COALESCE` in `stg__pluto`